lib/galaxy_test/selenium/test_history_panel_collections.py

```diff
@@ -132,6 +132,26 @@ def test_output_collection_states_running(self):
             for job in payload["jobs"]:
                 self.dataset_populator.cancel_job(job["id"])
 
+    @selenium_test
+    def test_collection_job_details(self):
+        ok_collection_hid, failed_collection_hid = self._generate_ok_and_failed_collections()
+        ok_collection_element = self.history_panel_wait_for_hid_state(ok_collection_hid, "ok")
+        failed_collection_element = self.history_panel_wait_for_hid_state(failed_collection_hid, "error")
+
+        ok_collection_element.collection_job_details_button.wait_for_and_click()
+        tool_id_component = self.components.job_details.galaxy_tool_id.wait_for_visible()
+        tool_exit_code_component = self.components.job_details.tool_exit_code.wait_for_visible()
+        self.screenshot("history_panel_collections_job_details_ok")
+        assert tool_id_component.text == "collection_creates_list"
+        assert int(tool_exit_code_component.text) == 0
+
+        failed_collection_element.collection_job_details_button.wait_for_and_click()
+        tool_id_component = self.components.job_details.galaxy_tool_id.wait_for_visible()
+        tool_exit_code_component = self.components.job_details.tool_exit_code.wait_for_visible()
+        self.screenshot("history_panel_collections_job_details_failed")
+        assert tool_id_component.text == "collection_creates_list_fail"
```

Comment:
I think the test would be more robust if you include this in the selector, so that `wait_for_visible` would actually wait for the text to change. What could happen here is that depending how quickly the rerender occurs you still get the old tool id in tool_id_component.text.

Reply:
Ok, I think I got it now. It took me a while to figure out that the tool id text was actually surrounded by a lot of spaces :sweat_smile:

Reply:
Thanks for taking that extra step, really appreciate it!